src/main/java/com/google/alpollo/model/SongEntity.java

```diff
@@ -26,4 +37,31 @@ public int compare(SongEntity a, SongEntity b) {
       return Double.compare(b.salience, a.salience);
     }
   };
+
+  @Override
+  public boolean equals(Object obj) {
+    if (obj == this) {
+      return true;
+    }
+
+    if (!(obj instanceof SongEntity)) { 
+      return false; 
+  } 
+
+    SongEntity entity = (SongEntity) obj;
+    return salience == entity.salience
+        && (Objects.equal(name,entity.name))
+        && (Objects.equal(type, entity.type))
+        && (Objects.equal(wikiLink,entity.wikiLink));
+  }
+
+  @Override
+  public int hashCode() {
+    int hash = 7;
+    hash = 31 * hash + (int) salience;
+    hash = 31 * hash + (name == null ? 0 : name.hashCode());
+    hash = 31 * hash + (type == null ? 0 : type.hashCode());
+    hash = 31 * hash + (wikiLink == null ? 0 : wikiLink.hashCode());
```

Comment:
If you want to be safer here, you can use Objects.hashcode() that will handle potential null values